dome9/resource_dome9_admission_control_policy.go

```diff
@@ -0,0 +1,121 @@
+package dome9
+
+import (
+	"github.com/dome9/dome9-sdk-go/dome9/client"
+	"github.com/dome9/dome9-sdk-go/services/admissioncontrol/admission_policy"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"log"
+)
+
+func resourceAdmissionPolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAdmissionControlPolicyCreate,
+		Read:   resourceAdmissionControlPolicyRead,
+		Update: resourceAdmissionControlPolicyUpdate,
+		Delete: resourceAdmissionControlPolicyDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"target_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"target_type": {
+				Type:         schema.TypeString,
+				Required:     true,
```

Comment:
whne checking the ac controller api model, i saw it is a required field with default value :
[Required] 
        public KubernetesAdmissionControlPolicyType TargetType { get; set; } = KubernetesAdmissionControlPolicyType.Environment;
 
 so i will do the same as our controller do, remains it required and verifying if it does not exists inside TF Configuration file under the policy resource , will set a default 'Environment' value, sounds good ?

Reply:
I don't see a reason for both. to my mind this means it is not required.
we might want to open a fix for controller to remove the required from TargetType since it is not required

Reply:
ok so i will change to optional here with existence verification